processing/src/main/java/org/apache/druid/frame/read/columnar/StringArrayFrameColumnReader.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.frame.read.columnar;
+
+import org.apache.datasketches.memory.Memory;
+import org.apache.druid.error.DruidException;
+import org.apache.druid.frame.Frame;
+import org.apache.druid.frame.write.columnar.FrameColumnWriters;
+import org.apache.druid.frame.write.columnar.StringFrameColumnWriter;
+import org.apache.druid.query.rowsandcols.column.Column;
+import org.apache.druid.query.rowsandcols.column.ColumnAccessorBasedColumn;
+import org.apache.druid.segment.ColumnValueSelector;
+import org.apache.druid.segment.column.BaseColumn;
+import org.apache.druid.segment.column.ColumnCapabilitiesImpl;
+import org.apache.druid.segment.column.ColumnType;
+import org.apache.druid.segment.data.ReadableOffset;
+
+/**
+ * Reader for {@link ColumnType#STRING_ARRAY}.
+ */
+public class StringArrayFrameColumnReader implements FrameColumnReader
+{
+  final int columnNumber;
+
+  /**
+   * Create a new reader.
+   *
+   * @param columnNumber column number
+   */
+  StringArrayFrameColumnReader(int columnNumber)
+  {
+    this.columnNumber = columnNumber;
+  }
+
+  @Override
+  public Column readRACColumn(Frame frame)
+  {
+    final Memory memory = frame.region(columnNumber);
+    validate(memory);
+
+    // String arrays always store multiple values per row.
+    assert isMultiValue(memory);
+    final long positionOfLengths = getStartOfStringLengthSection(frame.numRows());
+    final long positionOfPayloads = getStartOfStringDataSection(memory, frame.numRows());
+
+    StringFrameColumnReader.StringFrameColumn frameCol = new StringFrameColumnReader.StringFrameColumn(
```

Comment:
why isn't this using `StringArrayFrameColumn`? though i have some comments on that too...

Reply:
updated